src/Nethermind/Nethermind.JsonRpc.Test/Modules/AdminModuleTests.cs

```diff
@@ -0,0 +1,105 @@
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System.Collections.Generic;
+using FluentAssertions;
+using Nethermind.Blockchain;
+using Nethermind.Core;
+using Nethermind.Core.Test.Builders;
+using Nethermind.JsonRpc.Modules.Admin;
+using Nethermind.Network;
+using Nethermind.Network.Config;
+using Nethermind.Serialization.Json;
+using Nethermind.Stats.Model;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace Nethermind.JsonRpc.Test.Modules
+{
+    [TestFixture]
+    public class AdminModuleTests
+    {
+        private IAdminModule _adminModule;
+        private EthereumJsonSerializer _serializer;
+        private NetworkConfig _networkConfig;
+        private IBlockTree _blockTree;
+        private const string _enodeString = "enode://44826a5d6a55f88a18298bca4773fca5749cdc3a5c9f308aa7d810e9b31123f3e7c5fba0b1d70aac5308426f47df2a128a6747040a3815cc7dd7167d03be320d@127.0.0.1:30303";
+        
+        [SetUp]
+        public void Setup()
+        {
+            _blockTree = Build.A.BlockTree().OfChainLength(5).TestObject;
+            _networkConfig = new NetworkConfig();
+            IPeerManager peerManager = Substitute.For<IPeerManager>();
+            peerManager.ActivePeers.Returns(new List<Peer> {new Peer(new Node("127.0.0.1", 30303, true))});
+            
+            IStaticNodesManager staticNodesManager = Substitute.For<IStaticNodesManager>();
+            Enode enode = new Enode(_enodeString);
+            _adminModule = new AdminModule(_blockTree, _networkConfig, peerManager, staticNodesManager, enode, "C:\\dataDir");
+            _serializer = new EthereumJsonSerializer();
+        }
+        
+        [Test]
+        public void Test_node_info()
+        {
+            string serialized = RpcTest.TestSerializedRequest(_adminModule, "admin_nodeInfo");
+            JsonRpcSuccessResponse response = _serializer.Deserialize<JsonRpcSuccessResponse>(serialized);
+            JsonSerializerSettings settings = new JsonSerializerSettings();
+            settings.Converters = EthereumJsonSerializer.BasicConverters;
+            
+            NodeInfo nodeInfo = ((JObject) response.Result).ToObject<NodeInfo>(JsonSerializer.Create(settings));
+            nodeInfo.Enode.Should().Be(_enodeString);
+            nodeInfo.Id.Should().Be("44826a5d6a55f88a18298bca4773fca5749cdc3a5c9f308aa7d810e9b31123f3e7c5fba0b1d70aac5308426f47df2a128a6747040a3815cc7dd7167d03be320d");
```

Comment:
small comment: when doing object comparision you can create expectedNodeInfo object and use nodeInfo.Should().BeEquavilent(expectedNodeInfo) and it will go through inner properties and structure of object.

Reply:
ah yes, I was using it in some other places, here probably will not make much difference
I started using should / shouldbeequivalentto more after reviewing yours and Piotrek's tests